src/Components/Components/src/NavigationManager.cs

```diff
@@ -1,10 +1,14 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Buffers;
+using System.Linq;
```

Comment:
Is `System.Linq` needed? Pranav went on a whole campaign of removing these references from low-level code in the past.

Reply:
I guess it's for the `locationChangingTasks.Take` call. Might be possible to use an array segment or Span instead.

Reply:
Yeah, this was something I wasn't too happy about either. Unfortunately, `Task.WhenAll` only has overloads accepting `Task[]` and `IEnumerable<Task>`. So even if we use an `ArraySegment<Task>`, the extra allocation to get an `IEnumerable<Task>` seems inevitable. We could also consider not using `ArrayPool<Task>` at all and just allocating a `Task[]` with a hot path for the single handler scenario.

Reply:
> We could also consider not using `ArrayPool<Task>` at all and just allocating a `Task[]` with a hot path for the single handler scenario.

I've gone ahead and changed it to this - let me know if you disagree with that choice!